### PR TITLE
enhancement: add QA suite of tests

### LINKED ISSUE #1243: Create a suite of QA tests
- As a part of https://github.com/FuelLabs/fuel-indexer/issues/1226 we need to add some type of QA testing
- These QA tests would be different from unit tests (`#[test]`)  and integration tests (`forc index {do a thing}`) 
- QA tests would be run out-of-band from CI
  - _Ideally_ they would be run in order to ensure a release is OK to go out - but that's not a hard requirement for now
- QA tests are tests that:
  - Don't necessarily start at block 0
    - They start from arbitrary blocks 
    - Run much longer than tests on CI
      - Around ~10 minutes (rough guess)
      - Monitor various metrics of indexer performance
        - These various metrics include things like: CPU usage, memory usage, and indexer accuracy
          - Any significant change in any of these ☝🏽 metrics could potentially have a significant adverse effect on the block explorer
- If we're going to be shipping a production block explorer, we simply cannot afford to release bad code - therefore we need these QA tests

### PR DESCRIPTION
**Pull request by ra0x3**
Thanks for opening a PR with the Fuel Indexer project. Please review the **Checklist** below and ensure you've completed all of the necessary steps to make this PR review as painless as possible.

### Checklist
- [ ] Ensure your top-level commit message is in line with our [contributor guidelines](./CONTRIBUTING.md).
- [ ] Please add proper labels.
- [ ] If there is an issue associated with this PR, please link the issue (right-hand sidebar)
- [ ] If there is not an issue associated with this PR, add this PR to the "Fuel Indexer" project (right-hand sidebar)
- [ ] Please allow Codeowners at least 24 hours to do a first-pass review.
- [ ] Please add thoroughly detailed testing steps below.
- [ ] Please keep your Changelog message short and sweet.

### Description

- See the issue for more info
- Want to be clear that what's included in this PR is just _table stakes_ - this is not the final version
- After we add GraphQL query layer, we'll also include that in these tests
- There is no programmatic "action" for these yet - don't want to increase the scope
  - We can just run this QA program before we cut a release 

### Testing steps

- [ ] Build binaries locally
```bash
cargo build -p forc-index -p fuel-indexer -p fuel-indexer-api-server --release --locked
```
- [ ] Update QA script to point to your new `fuel-indexer`
```text
index 13700fce..b5a00e6f 100644
--- a/packages/fuel-indexer-benchmarks/src/bin/qa.rs
+++ b/packages/fuel-indexer-benchmarks/src/bin/qa.rs
@@ -491,7 +491,7 @@ async fn main() {
     let mani_path = explorer_root.join("fuel_explorer.manifest.yaml");

     // TODO: replace this with forc-index start after/during https://github.com/FuelLabs/fuel-indexer/pull/1348
-    let _proc = Command::new("fuel-indexer")
+    let _proc = Command::new("./target/release/fuel-indexer")
         .arg("run")
         .arg("--run-migrations")
         .arg("--fuel-node-host")
```
- [ ] Refresh DB
```bash 
bash scripts/utils/refresh_test_db.bash
```
- [ ] Run QA suite
```bash
cargo run -p fuel-indexer-benchmarks --bin qa -- --network beta-4.fuel.network --runs 10 --blocks 10000
```
- [ ] Confirm that your report looks [as expected](https://github.com/FuelLabs/fuel-indexer/files/12652289/Rashads-Air.lan-2023-09-18.indexer-qa.txt).

### Changelog

- enhancement: add QA suite of tests

### CODE REVIEW

File: packages/fuel-indexer/src/executor.rs [hunk 1]
```diff
@@ -90,23 +90,18 @@ pub fn run_executor<T: 'static + Executor + Send + Sync>(
         .unwrap_or(config.fuel_node.to_string());
 
     // Where should we initially start when fetching blocks from the client?
-    let mut cursor = executor.manifest().start_block().map(|x| {
```
Comment (ra0x3) [inline packages/fuel-indexer/src/executor.rs:93]: This is incorrect. We shouldn't be mutating any inputs. If the user want's to provide a certain `start_block` they should do that. If the user provides an invalid `start_block` just let it error out.

**Comment by ra0x3** [comment]
Example QA report:

[Rashads-Air.lan-2023-09-18.indexer-qa.txt](https://github.com/FuelLabs/fuel-indexer/files/12652289/Rashads-Air.lan-2023-09-18.indexer-qa.txt)

File: packages/fuel-indexer-api-server/src/uses.rs [hunk 2]
```diff
@@ -186,7 +186,7 @@ pub(crate) async fn remove_indexer(
     // unnecessary DatabaseError's appearing in the logs.
     tokio::time::sleep(std::time::Duration::from_secs(1)).await;
 
-    // Allways remove data when removing an indexer.
+    // Always remove data when removing an indexer.
```
Comment (lostman) [inline packages/fuel-indexer-api-server/src/uses.rs:189]: 👌

File: packages/fuel-indexer/src/executor.rs [hunk 3]
```diff
@@ -107,8 +107,10 @@ pub fn run_executor<T: 'static + Executor + Send + Sync>(
         panic!("Indexer({indexer_uid}) client node connection failed: {e}.")
     });
 
-    if end_block.is_none() {
-        warn!("No end_block specified in manifest. Indexer will run forever.");
+    if let Some(end_block) = end_block {
+        info!("Indexer({indexer_uid}) will stop at block #{end_block}.");
+    } else {
+        warn!("No end_block specified in manifest. Indexer({indexer_uid}) will run forever.");
```
Comment (lostman) [inline packages/fuel-indexer/src/executor.rs]:
```suggestion
        warn!("No end_block specified in the manifest. Indexer({indexer_uid}) will run forever.");
```

**Review by lostman — DISMISSED** [review]

**Review by lostman — APPROVED** [review]

**Review by deekerno — APPROVED** [review]
QA suite looks good to me. 👌

**Pull request merged**